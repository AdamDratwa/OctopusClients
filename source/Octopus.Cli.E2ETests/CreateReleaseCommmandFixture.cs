using System;
using System.IO;
using System.Reflection;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using Octopus.Cli.E2ETests.Util;
using Octopus.Cli.Util;

namespace Octopus.Cli.E2ETests
{
    [UseReporter(typeof(DiffReporter))]
    public class CreateReleaseCommmandFixture
    {
        [Test]
        public void PreventAutoChannelAndChannelArguments()
        {
            var octoexe = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().FullLocalPath()), "octo.exe");
            var args = new StringBuilder("create-release --server=https://localhost:8085 --apiKey=API-123 --project OctoFX ");
            var output = new StringBuilder();
            var result = SilentProcessRunner.ExecuteCommand(octoexe, args.Append("--autochannel --channel=Default").ToString(), Environment.CurrentDirectory, info => output.AppendLine(info), error => output.AppendLine(error));
            Approvals.Verify(output);
        }
    }
}
