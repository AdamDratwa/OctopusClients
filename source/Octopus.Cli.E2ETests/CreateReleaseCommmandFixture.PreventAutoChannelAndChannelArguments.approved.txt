Octopus Deploy Command Line Tool, version 3.3.13-autochannel.17+Branch.feature-autochannel.Sha.7b9f6f75342d02463f6b6762268b4e18a71de965

Handshaking with Octopus server: http://localhost:8065
Handshake successful. Octopus version: 3.3.5-bug-pkgscriptvar0015; API version: 3.0.0
Authenticated as: admin <> 
Cannot specify --channel and --autochannel arguments
Exit code: -1
